client/src/includes/sidePanel.js

```diff
@@ -60,7 +60,8 @@ export default function initSidePanel() {
         if (panel.hidden) {
           // eslint-disable-next-line no-param-reassign
           panel.hidden = false;
-          panel.dispatchEvent(new CustomEvent('show'));
+          // Don't fire the show event just yet,
+          // to ensure that the hide event for the other panels is fired first.
```

Comment:
I don't remember the exact details, but I found a problem with how the preview is activated by both the preview panel and the checks side panel on show.

When switching from panel A to B, the old side panel code dispatches the show event for panel B before the hide event for panel A. This would cause the auto update interval to no longer be set if you:
- open checks panel -> show event for checks panel -> interval is set
- switch to the preview panel
  - show event for preview panel -> interval is still active (or to be more precise, cleared and set again)
  - hide event for checks panel -> interval gets cleared
- make edits, no updates happen to the preview panel since the interval has been cleared

Similar thing if you do it the other way around. This is because we use the same handler for both panels, and we don't keep track which panel triggered it.

Besides, I think it's more logical to dispatch the hide events for the other panels first before dispatching the show event for the newly selected panel anyway.